packages/twenty-front/src/modules/workflow/components/WorkflowEditActionFormServerlessFunction.tsx

```diff
@@ -63,12 +89,26 @@ export const WorkflowEditActionFormServerlessFunction = (
     1_000,
   );
 
-  const handleInputChange = (key: string, value: any) => {
-    const newFunctionInput = { ...functionInput, [key]: value };
+  const handleInputChange = (value: any, path: string[]) => {
+    const newFunctionInput = { ...functionInput };
+    setNestedValue(newFunctionInput, path, value);
     setFunctionInput(newFunctionInput);
     updateFunctionInput(newFunctionInput);
   };
 
+  const getNestedValue = (obj: any, path: string[]) =>
+    path.reduce((o, key) => (o ? o[key] : undefined), obj);
+
+  // Utility to set nested value
+  const setNestedValue = (obj: any, path: string[], value: any) => {
```

Comment:
could this go into an util, as long as getNestedValue?

Reply:
yes